### PR TITLE
Add some notes about git mirrors

### PR DESCRIPTION
**Pull request by DrJosh9000**
This felt broadly useful, but I'm not 100% certain it belong here.

### CODE REVIEW

**Comment by pda** [comment]
Nice!

> You might guess that `git fetch` only ever adds new objects, but unfortunately,
> `git fetch` by default runs `git maintentance run --auto`, which sometimes
> cleans up old unreferenced objects.

Could we `git fetch --no-auto-maintenance`? Or is the corrupting fetch outside our control? Or is that maintenance there for a good reason? 😅

**Comment by DrJosh9000** [comment]
> Could we git fetch --no-auto-maintenance? Or is the corrupting fetch outside our control? Or is that maintenance there for a good reason? 😅

Supposing we run --no-auto-maintenance, then I imagine (assuming commits and branches continue churning objects remotely) the mirror would continue to grow until either Git becomes too slow to use, or disk fills up.

**Comment by pda** [comment]
> Supposing we run --no-auto-maintenance, then I imagine (assuming commits and branches continue churning objects remotely) the mirror would continue to grow until either Git becomes too slow to use, or disk fills up.

I can imagine use-cases where that's preferable to `git fetch` corrupting it, especially on CI agent machines which _probably_ have a finite lifetime before the mirror is thrown away. I imagine the growth of a typical non-garbage-collected git mirror isn't that much larger than a garbage-collected one, assuming most branches are merged and so most commits are kept forever. But there's probably details I'm not considering!

File: docs/git-mirror.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# How Git Mirrors Works
+
+## The goal
+
+The goal of Git mirrors is to:
+
+- reduce network bandwidth, and
+- reduce disk usage
+
+for machines running multiple agents side by side.
```
Comment (moskyb) [inline docs/git-mirror.md:10]: maaaaaaaaybe it's worth talking about how "git mirror on a ZFS share" is also a supported workflow?

**Review by moskyb — APPROVED** [review]
this is awesome awesome work - the inner workings of git mirrors have always been a bit of a mystery to me, but this cleared a whole bunch of stuff up

File: docs/git-mirror.md (hunk 1 above)
Comment (DrJosh9000) [reply docs/git-mirror.md]: Congratulations! You have won one (1) footnote!

**Comment by sj26** [comment]
Nice!

I have some really old hooks where I tried to use git worktrees for builds, if it's at all useful:

https://github.com/sj26/git-worktree-buildkite-hooks

**Pull request merged**